xarray/core/combine.py

```diff
@@ -626,15 +631,16 @@ def _combine_single_variable_hypercube(
     return concatenated
 
 
+# TODO remove empty list default param after version 0.19, see PR4696
 def combine_by_coords(
-    data_objects,
+    data_objects=[],
```

Comment:
```suggestion
    data_objects,
```

(It's [considered bad practice](https://nikos7am.com/posts/mutable-default-arguments/) to have mutable default arguments to functions in python.)

Reply:
I put this in because if someone calls this method with datasets as a named parameter the data_objects argument would be unspecified and their code would break with an unspecified argument error. This is part of the deprecation warning below.

Reply:
You make a good point, but that means the default argument should be `None`, not an empty list, as `None` is immutable.